mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -591,28 +623,72 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     if (labelType == 0) {
       // Regression
       ensembleType match {
-        case "RandomForest" =>
-          Left(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
-            "variance", treeDepth, maxBins, this.getRandomSeed))
-        case "GradientBoostedTrees" =>
+        case "mllib.RandomForest" =>
+          MLlibRFModel(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees,
+            featureSubsetStrategy, "variance", treeDepth, maxBins, this.getRandomSeed))
+        case "ml.RandomForest" =>
```

Comment:
For ml tests, I'd copy the TreeTests.setMetadata method from Spark and use it to set the categorical feature metadata.  Otherwise, the comparisons won't be fair.